### ISSUE TITLE
Provide option to disable search term highlighting

### ISSUE THREAD

**Issue by bleetsheep** [splitbrain/dokuwiki#3594]
The old bugs platform had this bug open: https://web.archive.org/web/20100127042248/http://bugs.splitbrain.org/index.php?do=details&task_id=1862

I don't know what happened to it but the whole domain seems to be gone without redirect, archive, or porting (luckily the internet archive stored it).

> I would like to suggest to add an option to disable search term highlighting.
> 
> Rationale: Most of the time I'm not really searching for a particular term but rather for a topic. In those cases, I feel that the (sometimes extensive) highlighting clutters the page display. If I'm really searching for a term, I can easily use the browser's find function. Getting rid of the highlighting (by modifying the URL) is much more difficult.
> 
> As far as I can tell, disabling search term highlighting would mean changing the search result page to create links *not* containing an "&s[]=..." extension.
> 
> I know about the workaround by adapting the color handling in conf/userstyle.css, see http://forum.dokuwiki.org/post/16778
> 
> Thanks for reading, Malte

---

I, too, would much prefer there not to be cluttered links after I click a search result, nor on-page clutter from word highlighting.

Dokuwiki seems to be the odd one out compared to other search engines and wiki engines. The default should probably be 'off'.

Perhaps this should be a wiki setting, perhaps a user preference?

**Comment by Klap-in** [comment]
The old bug tracker has moved to https://bugs.dokuwiki.org/1862.html

Is the nohighlight plugin working for you?
http://www.dokuwiki.org/plugin:nohighlight

**Comment by bleetsheep** [comment]
I am not an admin on the wiki, though I could suggest it. Thanks for the suggestion!

Is there a particular reason this is not just included in mainline? Since dokuwiki is quite unique with this behavior, it does not seem to me as though this should be the default. (Rather, the highlighting is extra code that should be an add-on. Or just an option.)

**Comment by bleetsheep** [comment]
> Is the nohighlight plugin working for you?

No. I actually got an admin to try to install it, but the last update of this plugin is from 2012 and it gives this error:

    [Wed Jan 19 10:10:42.321032 2022] [php7:error] [pid xyz] [client 192.169.1.2:51234] PHP Fatal error:  Declaration of action_plugin_nohighlight::register(&$controller) must be compatible with dokuwiki\Extension\ActionPlugin::register(Doku_Event_Handler $controller) in /path/to/dokuwiki/lib/plugins/nohighlight/action.php on line 36

Either way, I still stand by that this should be mainline and the special highlight code is what should be an extra plugin.
